site/public/js/server.js

```diff
@@ -1933,37 +1933,50 @@ function renderMarkdown(markdownContainer, url, content) {
 function addMarkdownCode(type) {
     const markdown_area = $(this).closest('.markdown-area');
     const markdown_header = markdown_area.find('.markdown-area-header');
-    //don't allow markdown insertion if we are in preview mode
+    // Don't allow markdown insertion if we are in preview mode
     if (markdown_header.attr('data-mode') === 'preview') {
         return;
     }
 
-    const cursor = $(this).prop('selectionStart');
+    const start = $(this).prop('selectionStart');
+    const end = $(this).prop('selectionEnd');
     const text = $(this).val();
     let insert = '';
+    const selectedText = text.substring(start, end);
+
     switch (type) {
         case 'code':
-            if (text.substring(text.substring(0, cursor).split('').lastIndexOf('\n'), cursor).length !== 1) {
+            if (text.substring(text.substring(0, start).split('').lastIndexOf('\n'), start).length !== 1) {
                 insert = '\n';
             }
-            insert += '```\ncode\n```\n';
+            insert += `\`\`\`\n${selectedText}\n\`\`\`\n`;
```

Comment:
Cursor is placed weirdly when starting code 
![image](https://github.com/Submitty/Submitty/assets/34077605/e8762aa8-e5e4-4f7c-bc87-9046de9f6734)